src/js/Application.jsx

```diff
@@ -411,14 +420,117 @@ export default class Application extends Component {
     }
 
     let footerStyle = this.state.showFooter ? "footroom-wrapper" : "footroom-wrapper__hide";
+
+    // Determine the headroom space at the top of the scrollable pane (not related to headroom.js)
+    // console.log("Determine the headroom space pathname:" + pathname);
     let appBaseClass = "app-base";
-    if (isCordova()) {
-      if (!hasIPhoneNotch()) {
-        appBaseClass += " cordova-base-iphonex";
+    if (isWebApp()) {
+      appBaseClass += " headroom-webapp";
+    } else {
+      appBaseClass += " cordova-base";
+      if (isIOS()) {
+        appBaseClass += " headroom-ios";
+        if (hasIPhoneNotch()) {
+          appBaseClass += "-notch";
+        } else if (isIPhone678()) {
+          appBaseClass += "-678";
+        } else if (isIPhone678Plus()) {
+          appBaseClass += "-678plus";
+        } else {  // iPad
+          appBaseClass += "-ipad";
+        }
       } else {
-        appBaseClass += " cordova-base";
+        appBaseClass += " headroom-android";
+        appBaseClass += getAndroidSize();
+      }
+    }
+    if (stringContains("/ballot", pathname)) {
+      appBaseClass += "-secondary";
+    } else if (stringContains("/candidate/", pathname) ||
+      (stringContains("/settings/", pathname) && isCordova()) ||
+      stringContains("/measure/", pathname)) {
+      appBaseClass += "-backto";
+    } else {
+      appBaseClass += "-full";
+    }
+    // console.log("Determine the headroom space classname:" + appBaseClass);
+    // End of headroom space determination
+
+
+    // Determine the headroom space at the top of the scrollable pane (not related to headroom.js)
+    // console.log("Determine the headroom space pathname:" + pathname);
+    let isSecondary = stringContains("/ballot", pathname);
+    let isBackto =
+      stringContains("/candidate/", pathname) ||
+      (stringContains("/settings/", pathname) && isCordova()) ||
+      stringContains("/measure/", pathname);
+    let isFull = !(isSecondary || isBackto);
+    let androidSize = getAndroidSize();
+    // let appBaseClass = "";
+
+    if (isWebApp()) {
+      if (isSecondary) {
+        appBaseClass = "app-base headroom-webapp-secondary";
+      } else if (isBackto) {
+        appBaseClass = "app-base headroom-webapp-backto";
+      } else {
+        appBaseClass = "app-base headroom-webapp-full";
+      }
+    } else if (isAndroid()) {
+      if (androidSize === "-xl" && isSecondary) {
+        appBaseClass = "app-base cordova-base headroom-android-xl-secondary";
+      } else if (androidSize === "-lg" && isSecondary) {
```

Comment:
Repeating "app-base cordova-base " isn't necessary. You could do this following code and it would be fine, since there is a space between "cordova-base" and "headroom-android-xl-secondary":
} else if (isAndroid()) {
      appBaseClass = "app-base cordova-base";
      if (androidSize === "-xl" && isSecondary) {
        appBaseClass += " headroom-android-xl-secondary";